snakemake/cli.py

```diff
@@ -1026,20 +1027,26 @@ def get_argument_parser(profiles=None):
         action="store_true",
         help="Show available target rules in given Snakefile.",
     )
-    group_utils.add_argument(
+    group_exec.add_argument(
         "--dag",
-        action="store_true",
+        nargs="?",
+        choices=PrintDag.choices(),
+        default=set(),
+        parse_func=PrintDag.parse_choices_set,
         help="Do not execute anything and print the directed "
-        "acyclic graph of jobs in the dot language. Recommended "
+        "acyclic graph of jobs in the dot language or in mermaid-js. Recommended "
         "use on Unix systems: snakemake `--dag | dot | display`. "
         "Note print statements in your Snakefile may interfere "
         "with visualization.",
     )
     group_utils.add_argument(
         "--rulegraph",
-        action="store_true",
+        nargs="?",
+        choices=PrintDag.choices(),
+        default=set(),
```

Comment:
Shouldn't the default be one of the two choices? I would say dot in order to maintain backwards compatibility. (same above for --dag)